docs/wcs/index.rst

```diff
@@ -28,89 +42,139 @@ It performs three separate classes of WCS transformations:
            WCS transformation from pixel to world coordinates.
 
-       - From world to pixel coordinates:
+    2. From world to pixel coordinates:
 
-         - `~astropy.wcs.wcs.WCS.all_world2pix`: Perform all three
+        - `~astropy.wcs.wcs.WCS.all_world2pix`: Perform all three
            transformations (core WCS, SIP and table lookup
            distortions) from world to pixel coordinates, using an
            iterative method if necessary.
 
-         - `~astropy.wcs.wcs.WCS.wcs_world2pix`: Perform just the core
+        - `~astropy.wcs.wcs.WCS.wcs_world2pix`: Perform just the core
            WCS transformation from world to pixel coordinates.
 
-       - Performing `SIP`_ transformations only:
+    3. Performing `SIP`_ transformations only:
 
-         - `~astropy.wcs.wcs.WCS.sip_pix2foc`: Convert from pixel to
+        - `~astropy.wcs.wcs.WCS.sip_pix2foc`: Convert from pixel to
            focal plane coordinates using the `SIP`_ polynomial
            coefficients.
 
-         - `~astropy.wcs.wcs.WCS.sip_foc2pix`: Convert from focal
+        - `~astropy.wcs.wcs.WCS.sip_foc2pix`: Convert from focal
            plane to pixel coordinates using the `SIP`_ polynomial
            coefficients.
 
-       - Performing `distortion paper`_ transformations only:
+    4. Performing `distortion paper`_ transformations only:
 
-         - `~astropy.wcs.wcs.WCS.p4_pix2foc`: Convert from pixel to
+        - `~astropy.wcs.wcs.WCS.p4_pix2foc`: Convert from pixel to
            focal plane coordinates using the table lookup distortion
            method described in the FITS WCS `distortion paper`_.
 
-         - `~astropy.wcs.wcs.WCS.det2im`: Convert from detector
+        - `~astropy.wcs.wcs.WCS.det2im`: Convert from detector
            coordinates to image coordinates.  Commonly used for narrow
            column correction.
 
-For example, to convert pixel coordinates from a two dimensional image to world coordinates::
+For example, to convert pixel coordinates from a two dimensional image
+to world coordinates::
 
     >>> from astropy.wcs import WCS
-    >>> w = WCS('image.fits')  # doctest: +IGNORE_WARNINGS
-    >>> lon, lat = w.all_pix2world(30, 40, 0)
+    >>> wcs = WCS('image.fits')  # doctest: +IGNORE_WARNINGS
+    >>> lon, lat = wcs.all_pix2world(30, 40, 0)
     >>> print(lon, lat)
     31.0 41.0
 
-
-Using `astropy.wcs`
-===================
+The applications of the other transformations in the above list are similar.
 
 Loading WCS information from a FITS file
 ----------------------------------------
-
 This example loads a FITS file (supplied on the commandline) and uses
 the WCS cards in its primary header to transform.
 
 .. literalinclude:: examples/from_file.py
    :language: python
 
-Building a WCS structure programmatically
------------------------------------------
+Using `astropy.wcs`
+===================
+
+High level interface
+--------------------
+
+Low level interface
+-------------------
```

Comment:
The rename this section to `Legacy Interface` and move the current `Simple example` here.